src/main/java/components/Parser.java

```diff
@@ -0,0 +1,229 @@
+package components;
+
+import command.*;
+import exception.*;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+import task.Task.TaskType;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a handler that parses user input.
+ */
+public class Parser {
+    /**
+     * Returns a Command object that can execute the user input.
+     *
+     * @param input Input by user.
+     * @return Command to be executed based on input.
+     * @throws NyabotException If exception arises during parsing. Examples include
+     * NyabotNoSuchCommandException when there is no relevant command.
+     */
+    public static Command parse(String input) throws NyabotException {
+        String firstWord = input.split(" ")[0];
+        switch(firstWord) {
+        case "bye":
+            return new ExitCommand();
```

Comment:
I like how you didn't indent the `case` statements in your switch-case, following the coding standards. Good job